depscan/lib/package_query/cargo_pkg.py

```diff
@@ -57,31 +60,32 @@ def cargo_pkg_risk(pkg_metadata, is_private_pkg, scope, pkg):
     ):
         is_deprecated = True
     latest_deprecated = False
-    version_nums = list(versions_dict.keys())
-    try:
-        first_version_num = min(
-            version_nums
-        )
-        latest_version_num = max(
-            version_nums
-        )
-    except (ValueError, TypeError):
-        # First version number is latest, while last is the oldest release.
-        first_version_num = version_nums[-1]
-        latest_version_num = version_nums[0]
-    first_version = versions_dict.get(first_version_num)
-    latest_version = versions_dict.get(latest_version_num)
+
+    # Logic fails for version like 2.0.9 > 2.0.14 in python
```

Comment:
Can you use semver comparison? Example: https://github.com/owasp-dep-scan/dep-scan/blob/master/depscan/lib/package_query/pypi_pkg.py#L58

Reply:
cargo api does sort it with 0 being the latest and the last being the oldest version.
should I still add the logic?

Reply:
Yes, good idea to use our logic since sorting a sorted list will be faster